### ISSUE TITLE
Emulators start and suddenly shutting down

### ISSUE THREAD

**Issue by katopz** [firebase/firebase-tools#3615]
### [REQUIRED] Environment info

**firebase-tools:** 9.16.0

**Platform:** macOS 11.4

### [REQUIRED] Test case
I've 2 mac which `10.15.7` working fine but `11.4` refuse to start emulator.

### [REQUIRED] Steps to reproduce
```
firebase emulators:start
```

### [REQUIRED] Expected behavior
emulator start

### [REQUIRED] Actual behavior

```
i  emulators: Starting emulators: functions, firestore
i  emulators: Shutting down emulators.
i  hub: Stopping emulator hub
:warning:  functions: Port 5001 is not open on localhost, could not start Functions Emulator.
:warning:  functions: To select a different host/port, specify that host/port in a firebase.json config file:
      {
        // ...
        "emulators": {
          "functions": {
            "host": "HOST",
            "port": "PORT"
          }
        }
      }
i  emulators: Shutting down emulators.

Error: Could not start Functions Emulator, port taken.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by JasonHoku** [comment]
Do you have a 3rd party firewall or antivirus software that could be blocking applications or ports?

There is also the native firewall settings to check:
https://support.apple.com/guide/mac-help/change-firewall-preferences-on-mac-mh11783/mac

**Comment by katopz** [comment]
@JasonHoku not any, native firewall is turn off by default. And I don't thing it will cause `emulators: Shutting down emulators.` at step 2 anyhow.

**Comment by JasonHoku** [comment]
> @JasonHoku not any, native firewall is turn off by default. And I don't thing it will cause `emulators: Shutting down emulators.` at step 2 anyhow.

Yes but the error warnings:
:warning:  functions: Port 5001 is not open on localhost, could not start Functions Emulator.
and
Error: Could not start Functions Emulator, port taken.

Are saying the issue is because it cannot open 5001

Are you possibly running two functions emulators apps simultaneously?  If so the emulator does not automatically assign a new port if taken and you'll have to designate it in the firebase.json or re-init the project.

**Comment by katopz** [comment]
I think so at first, so I try `lsof -ti:5001` with no result.
To make sure, I `kill` all related port I can think of.
And even reboot and try again with no luck.
Retry `firebase init` didn't work either.

Anyway when I try change function port to `9991`.
I got 
```
i  emulators: Starting emulators: functions, firestore
:warning:  functions: The following emulators are not running, calls to these services from the Functions emulator will affect production: auth, database, hosting, pubsub, storage
:heavy_check_mark:  functions: Using node@16 from host.
i  firestore: Firestore Emulator logging to firestore-debug.log
:warning:  firestore: Fatal error occurred: 
   Firestore Emulator has exited with code: 1, 
   stopping all running emulators
i  functions: Stopping Functions Emulator
i  firestore: Stopping Firestore Emulator
:warning:  firestore: Error stopping Firestore Emulator
i  hub: Stopping emulator hub
```

Which is fine because I've`firestore-debug.log` to investigate now and it said.
```
The operation couldn’t be completed. Unable to locate a Java Runtime.
Please visit http://www.java.com/ for information on installing Java.
```

Cool, This is new mac which I didn't install java runtime yet so I install it and try again.
And now it's working!

But, I'm still curious about `5001` port which no one using it so I switch back to `5001` and try again... didn't work!
I've to stick with other port number then.

## TLDR
- `firestore-debug.log` not create and emu shutting down when no `java` installed.
- port `5001` didn't work for some reason even no one using it.
- Error detail seem incorrect about port taken?

I'll leave issue open and let the team desire to close it as is or fix or investigate it somehow.

Thanks

**Comment by galaxyblur** [comment]
Just experienced this on a new Mac and it would definitely help if this error ("Unable to locate a Java Runtime.") was printed to the console instead. Additionally, I don't see any mention of Java in the emulator docs.
